types/errors/errors.go

```diff
@@ -394,3 +397,36 @@ type causer interface {
 type unpacker interface {
 	Unpack() []error
 }
+
+func GRPCWrap(err error, c codes.Code, msg string) error {
+	if err != nil {
+		st := status.New(c, msg)
+		var sdkErr *Error
+		if errors.As(err, &sdkErr) {
+			errorInfo := &errdetails.ErrorInfo{
+				Reason:   sdkErr.Error(),
+				Metadata: map[string]string{"Codespace": sdkErr.Codespace(), "ABCICode": fmt.Sprintf("%d", sdkErr.ABCICode())},
+			}
+			st, err = st.WithDetails(errorInfo)
+			if err != nil {
+				return err
+			}
```

Comment:
this err is overwriting the true error that we want to wrap, I don't think we should to do this
```suggestion
			st, withDetailsErr := st.WithDetails(errorInfo)
			if withDetailsErr != nil {
			         // log withDetailsErr? we could return a multi error here
				return err
			}
```

Reply:
Regarding this, I prefer to avoid to hide withDetailsError, because it's something it will be detected during implementation. It's rare it will be triggered in prod